File: SixUiViews/app/src/main/java/cn/six/open/view/rv/OneAdapter/RvViewHolder.java
package cn.six.open.view.rv.OneAdapter;

import android.support.v4.util.SparseArrayCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

/**
 * Created by songzhw on 2016-08-12
 */
public class RvViewHolder extends RecyclerView.ViewHolder {
    private SparseArrayCompat<View> views;
    private View convertView;

    private RvViewHolder(View itemView) {
        super(itemView);
        this.convertView = itemView;
        views = new SparseArrayCompat<View>();
    }

    public static RvViewHolder createViewHolder(View itemView) {
        RvViewHolder holder = new RvViewHolder(itemView);
        return holder;
    }

    public static RvViewHolder createViewHolder(ViewGroup parent, int layoutId) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(layoutId, parent, false);
        RvViewHolder holder = new RvViewHolder(itemView);
        return holder;
    }

    public <T extends View> T getView(int id){
        View view = views.get(id);
        if(view == null){
            view = convertView.findViewById(id);
            views.put(id, view);
        }
        return (T) view;
    }

    // ============================================
    public void setText(int id, String str){
        TextView tv = getView(id);
        tv.setText(str);
    }

    public void setSrc(int id, int drawId){
        ImageView iv = getView(id);
        iv.setImageResource(drawId);
    }

    public void setBackground(int id, int bgResId){
        View view = getView(id);
        view.setBackgroundResource(bgResId);
    }


}

File: SixUiViews/lib/src/main/java/ca/six/views/rv/RvViewHolder.kt
package ca.six.views.rv


import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.collection.SparseArrayCompat
import androidx.recyclerview.widget.RecyclerView

class RvViewHolder private constructor(private val convertView: View) :
    RecyclerView.ViewHolder(convertView) {
    private val views: SparseArrayCompat<View> = SparseArrayCompat()

    fun <T : View> getView(id: Int): T {
        var view: View? = views.get(id)
        if (view == null) {
            view = convertView.findViewById(id)
            views.put(id, view)
        }
        return view as T
    }

    // ============================================
    fun setText(id: Int, str: String) {
        val tv = getView<TextView>(id)
        tv.text = str
    }

    fun setSrc(id: Int, drawId: Int) {
        val iv = getView<ImageView>(id)
        iv.setImageResource(drawId)
    }

    fun setBackground(id: Int, bgResId: Int) {
        val view = getView<View>(id)
        view.setBackgroundResource(bgResId)
    }

    fun setClickListener(id: Int, listener: View.OnClickListener) {
        val view = getView<View>(id)
        view.setOnClickListener(listener)
    }

    fun setVisibility(id: Int, visibility: Int) {
        val view = getView<View>(id)
        view.visibility = visibility
    }


    companion object {
        fun createViewHolder(itemView: View): RvViewHolder {
            return RvViewHolder(itemView)
        }

        fun createViewHolder(parent: ViewGroup, layoutId: Int): RvViewHolder {
            val itemView = LayoutInflater.from(parent.context).inflate(layoutId, parent, false)
            return RvViewHolder(itemView)
        }
    }

}
